### ISSUE TITLE
Convert grunt entry points to support Typescript

### ISSUE THREAD

**Issue by samreid** [phetsims/chipper#1464]
In https://github.com/phetsims/chipper/issues/1459 we established how to use TypeScript in chipper via spawned processes. We would like to expand that usage to perennial/aqua/dot/kite/scenery and within chipper/. Be aware of https://github.com/phetsims/chipper/issues/1463 which should likely happen before this.

- [x] Factor out util/ style components like execTask, registerTasks, spawn, etc in chipper/js/Gruntfile.js in a way that can be leveraged by other repo gruntfiles. 
- [x] Move grunt utility files like gruntSpawn, registerTasks, etc. to perennial-alias.
- [x] Move perennial/grunt tasks to /tasks (new issue)
- [ ] Move dot/kite/scenery grunt tasks to tasks/ (new issue)

### repos with a typescript_chipper_1464 branch
- [x] DELETED chipper
- [x] DELETED aqua
- [x] DELETED dot

**Comment by samreid** [comment]
<details>

```diff

Subject: [PATCH] Remove debug output, see https://github.com/phetsims/chipper/issues/1464
---
Index: js/grunt/tasks/default.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/grunt/tasks/default.ts b/js/grunt/tasks/default.ts
new file mode 100644
--- /dev/null	(date 1726665967088)
+++ b/js/grunt/tasks/default.ts	(date 1726665967088)
@@ -0,0 +1,29 @@
+// Copyright 2024, University of Colorado Boulder
+
+/**
+ * Erases the build/ directory and all its contents, and recreates the build/ directory
+ *
+ * @author Sam Reid (PhET Interactive Simulations)
+ */
+
+import * as grunt from 'grunt';
+import { lintAll } from './lint-all';
+
+( async () => {
+  if ( grunt.option( 'lint' ) === false ) {
+    // do nothing
+  }
+  else {
+    await lintAll();
+  }
+
+  if ( grunt.option( 'report-media' ) === false ) {
+    // do nothing
+  }
+  else {
+    require( './report-media' );
+  }
+
+  require( './clean' );
+  require( './build' );
+} )();
\ No newline at end of file
Index: js/grunt/tasks/lint-all.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/grunt/tasks/lint-all.ts b/js/grunt/tasks/lint-all.ts
--- a/js/grunt/tasks/lint-all.ts	(revision 9450827abc96e869f16ddf0c16df33c71b0ff795)
+++ b/js/grunt/tasks/lint-all.ts	(date 1726666208980)
@@ -1,12 +1,11 @@
 // Copyright 2024, University of Colorado Boulder
 
 /**
- * lint all js files that are required to build this repository (for the specified brands)
+ * Lints all JS files required to build this repository (for the specified brands)
  *
  * @author Sam Reid (PhET Interactive Simulations)
  */
 
-// @ts-expect-error
 import assert from 'assert';
 import * as grunt from 'grunt';
 import buildLocal from './util/buildLocal';
@@ -22,21 +21,35 @@
 const cache = !getOption( 'disable-eslint-cache' );
 const fix = getOption( 'fix' );
 const chipAway = getOption( 'chip-away' );
-assert && assert( !getOption( 'patterns' ), 'patterns not support for lint-all' );
+assert( !getOption( 'patterns' ), 'patterns not support for lint-all' );
 
 const getPhetLibs = require( '../getPhetLibs' );
 
 const brands = getBrands( grunt, repo, buildLocal );
 
-( async () => {
+/**
+ * Executes the linting process.
+ */
+export async function lintAll(): Promise<void> {
   const lintReturnValue = await lint( getPhetLibs( repo, brands ), {
     cache: cache,
     fix: fix,
     chipAway: chipAway
   } );
 
-// Output results on errors.
+  // Output results on errors.
   if ( !lintReturnValue.ok ) {
     grunt.fail.fatal( 'Lint failed' );
   }
-} )();
\ No newline at end of file
+  else {
+    console.log( 'Linting completed successfully.' );
+  }
+}
+
+// Detect if the script is run directly
+if ( import.meta.url === `file://${process.argv[ 1 ]}` ) {
+  lintAll().catch( error => {
+    console.error( 'Linting failed:', error );
+    process.exit( 1 );
+  } );
+}
\ No newline at end of file

**Comment by zepumph** [comment]
I made https://github.com/phetsims/chipper/issues/1465 for the less interesting commits that are just about TypeScript conversion.

**Comment by zepumph** [comment]
Working on this now.

**Comment by zepumph** [comment]
Made a commit on perennial today. This isn't exactly a "commit point" as about 75% of the tasks still aren't converted, but are in their new files. I'll keep cranking away (good thing we are on the development branch).

**Comment by zepumph** [comment]
Ok. We are making good coverage here, but the main point of this issue is to convert Gruntfiles into the sub-task pattern so that we have no constaint on Grunt for modules we want to start converting to typescript. Here is the list of gruntfiles that most likely leak into perennial dependencies and so we want to use the new pattern on:

- [x] chipper
- [x] perennial
- [x] aqua
- [ ] dot
- [ ] kite
- [ ] quake\js
- [ ] rosetta
- [ ] skiffle\js
- [ ] weddell
- [ ] Move js/Gruntfile to top level in all repos

From here we will be largely unblocked for converting to Typescript.

**Comment by zepumph** [comment]
All items have been handled or are now in side issues. Closing
